jumanji/environments/combinatorial/routing/rb_board_generator_v1.py

```diff
@@ -0,0 +1,319 @@
+            if self.layout[head.x, head.y]:
+                invalid_head = True
+            else:
+                position = Vector(head.x, head.y)
+                dir_primary, dir_second = self.get_wiring_directions(head)
+                num_steps = max(self.dim.x, self.dim.y)
+                for step in range(num_steps):
+                    # Check for valid step in primary direction
+                    if (0 <= (position.x + dir_primary.x) < self.dim.x) \
+                      and (0 <= (position.y + dir_primary.y) < self.dim.y)\
+                      and not self.layout[position.x+dir_primary.x, position.y+dir_primary.y]:
+                        position.x += dir_primary.x
+                        position.y += dir_primary.y
+                        self.layout[position.x, position.y] = 3*self.num_wires + WIRE 
+                        invalid_head = False
+                    # Else check for valid step in second direction
+                    elif (0 <= position.x + dir_second.x < self.dim.x) \
+                      and (0 <= position.y + dir_second.y < self.dim.y)\
+                      and not self.layout[position.x+dir_second.x, position.y+dir_second.y]:
+                        position.x += dir_second.x
+                        position.y += dir_second.y
+                        self.layout[position.x, position.y] = 3*self.num_wires + WIRE 
+                        invalid_head = False
+        # Mark the head and target cells
+        self.layout[head.x, head.y] = 3*self.num_wires + HEAD
+        self.layout[position.x, position.y] = 3*self.num_wires + TARGET
+        self.num_wires += 1
+        return None
+
+
+    def is_full(self) -> bool:
+        # Return a boolean if there is no room to fit any more wires on the board
+        for i in range(self.dim.x):
+            for j in range(self.dim.y):
+                # Return False if there are any adjacent open spots
+                if (i < self.dim.x-1):
+                    if (self.layout[i, j] == 0) and (self.layout[i+1, j] == 0):
+                        return False
+                if (j < self.dim.y-1):
+                    if (self.layout[i, j] == 0) and (self.layout[i, j+1] == 0):
+                        return False
+        # Return True if there were no adjacent open spots
+        return True
+
+    def connectible_cells(self, x_head: int, y_head: int) -> list:
+        # Return a list of 2D tuples, cells that are connectible to (x_head, y_head)
+        connectible_list = []
+        self.add_connectible_cell(x_head, y_head, connectible_list)
+        return connectible_list
+
+    def add_connectible_cell(self, x_pos: int, y_pos: int, connectible_list: list) -> list:
+        """ Add the specified cell to the list, recursively call adjacent cells, and return list
+
+        ARGS:
+            x_pos, y_pos (int, int) : cell to add to the list
+            connectible list (list[(int,int)...] : input list of connected cells
```

Comment:
In python 3.8 typing still uses capital letters as so: List[int]. You will also need to import List from Typing